src/helpers.js

```diff
@@ -117,25 +119,88 @@ export function isTraversable(tileID, me) {
   }
 }
 
+/* eslint no-param-reassign: ["error", { "ignorePropertyModificationsFor": ["bigGrid"] }] */
+
+/*
+ * place all values from smallGrid into bigGrid. Align the upper left corner at x, y
+ */
+export function fillGridWithSubgrid(bigGrid, smallGrid, x, y) {
+  const smallGridWidth = smallGrid.length;
+  const smallGridHeight = smallGrid[0].length;
+
+  for (let i = 0; i < smallGridWidth; i++) {
+    for (let j = 0; j < smallGridHeight; j++) {
+      bigGrid[i + x][j + y] = smallGrid[i][j];
+    }
+  }
+}
+
+
+/* Returns a 2d cell array of traversible (1) and blocked (0) cells inside a tile.
+ *
+ * tileIsTraversable: if this tile is traversable
+ * cpt: number of cells per tile
+ * objRadius: radius of untraversable object in pixels
+ */
+export function traversableCellsInTile(tileIsTraversable, cpt, objRadius) {
+  const gridTile = [];
+  // Start with all traversable
+  let i;
+  let j;
+  for (i = 0; i < cpt; i++) {
+    gridTile[i] = new Array(cpt);
+    for (j = 0; j < cpt; j++) {
+      gridTile[i][j] = 1;
+    }
+  }
+
+  if (!tileIsTraversable) {
```

Comment:
Ah. Yep. Didn't even consider that use case. We can probably for now just assume diagonal walls are completely non-traversable by passing in an object radius of 29 or greater. In fact, now that I think about it, there's no good way using this function to handle ANY walls, and right now we're just assuming walls are non-traverable circles with radius 20. The bot behaves almost the exact same under that assumption, but this is definitely really bad.

It seems like a reasonably difficult fix. @altodyte should we fix this before merging this PR, or open an issue for it and deal with it later? (Right now, the bot plays just as well as it did before, so the code *works*)

Personally I think we should address it next sprint to keep moving forward. I'll open the issue in our backlog and leave it up to you do prioritize it.